### PR TITLE
chore(plugin-garfish): compatible react-router-v6

### PR DESCRIPTION
**Pull request by zhoushaw**
# PR Details

* chore(plugin-garfish): compatible react-router-v6

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/runtime/plugin-garfish/package.json [hunk 1]
```diff
@@ -70,7 +71,7 @@
     "react-loadable": "^5.5.0"
   },
   "peerDependencies": {
-    "@modern-js/runtime": "workspace:^2.0.0-beta.4"
+    "@modern-js/runtime": "workspace:*"
```
Comment (caohuilin) [inline packages/runtime/plugin-garfish/package.json:74]: peerDependencies 不能写 workspace:*

File: packages/runtime/plugin-garfish/src/runtime/utils/apps.tsx [hunk 2]
```diff
@@ -96,79 +115,64 @@ function getAppInstance(
...
+          } else {
+            logger(`MicroApp Garfish.loadApp "${appInfo.name}" mount`, {
+              appInfo: appInstance.appInfo,
+              appInstance,
+            });
+            await appInstance?.mount();
+          }
+        } catch (error) {
+          setLoadingState({
+            isLoading: true,
```
Comment (KyrieLii) [inline packages/runtime/plugin-garfish/src/runtime/utils/apps.tsx:148]: error 也保持 loading 态吗

File: tests/integration/garfish/main-router-v6/src/modern-app-env.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/// <reference types='@modern-js/app-tools/types' />
+/// <reference types='@modern-js/plugin-testing/type' />
+/// <reference types='@modern-js/plugin-garfish/type' />
+/// <reference types='@modern-js/plugin-express/types' />
+/// <reference types='@modern-js/plugin-koa/types' />
+/// <reference types='@modern-js/plugin-router-v5/types' />
```
Comment (KyrieLii) [inline tests/integration/garfish/main-router-v6/src/modern-app-env.d.ts:6]: v6 路由不需要这个

File: tests/integration/garfish/main-router-v6/src/routes/user/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { Outlet, useLoaderData, useMatches } from '@modern-js/runtime/router';
+import { useModuleApps } from '@modern-js/plugin-garfish/runtime';
```
Comment (KyrieLii) [inline tests/integration/garfish/main-router-v6/src/routes/user/layout.tsx:2]: 这里可以从 `@modern-js/runtime/garfish` 这里导出 ？ ( 我不确定

File: packages/runtime/plugin-garfish/src/runtime/utils/apps.tsx (hunk 2 above)
Comment (zhoushaw) [reply packages/runtime/plugin-garfish/src/runtime/utils/apps.tsx:148]: error 的时候展示 loading 的 error 状态

File: tests/integration/garfish/main-router-v6/src/routes/user/layout.tsx (hunk 4 above)
Comment (zhoushaw) [reply tests/integration/garfish/main-router-v6/src/routes/user/layout.tsx:2]: 这两种方式都可以，不过就是之前用户使用 @modern-js/runtime/garfish 这种用法经常类型提示异常

Comment (zhoushaw) [reply tests/integration/garfish/main-router-v6/src/routes/user/layout.tsx]: 不过 @modern-js/runtime/garfish 更短些，貌似这样会更好我改成 @modern-js/runtime/garfish 吧

**Review by nyqykk — DISMISSED** [review]

**Pull request merged**